parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReader.java

```diff
@@ -28,5 +30,5 @@
 
   List<TripleIterator<?>> columns();
 
-  void setPageSource(PageReadStore pageStore, long rowPosition);
+  void setPageSource(PageReadStore pageStore, long rowPosition, Optional<RowRanges> rowRanges);
```

Comment:
>   java.method.numberOfParametersChanged: The number of parameters of the method have changed.
>   
>   old: method void org.apache.iceberg.parquet.ParquetReader<T>::<init>(org.apache.iceberg.io.InputFile, org.apache.iceberg.Schema, org.apache.parquet.ParquetReadOptions, java.util.function.Function<org.apache.parquet.schema.MessageType, org.apache.iceberg.parquet.ParquetValueReader<?>>, org.apache.iceberg.mapping.NameMapping, org.apache.iceberg.expressions.Expression, boolean, boolean)
>   new: method void org.apache.iceberg.parquet.ParquetReader<T>::<init>(org.apache.iceberg.io.InputFile, org.apache.iceberg.Schema, org.apache.parquet.ParquetReadOptions, java.util.function.Function<org.apache.parquet.schema.MessageType, org.apache.iceberg.parquet.ParquetValueReader<?>>, org.apache.iceberg.mapping.NameMapping, org.apache.iceberg.expressions.Expression, boolean, boolean, boolean)
>   
>   SOURCE: BREAKING, BINARY: BREAKING
>   
>   From old archive: iceberg-parquet-1.1.0.jar
>   From new archive: iceberg-parquet-3e8b750.jar
>   
>   If this is an acceptable break that will not harm your users, you can ignore it in future runs like so for:
>   
>     * Just this break:
>         ./gradlew :iceberg-parquet:revapiAcceptBreak --justification "{why this break is ok}" \
>           --code "java.method.numberOfParametersChanged" \
>           --old "method void org.apache.iceberg.parquet.ParquetReader<T>::<init>(org.apache.iceberg.io.InputFile, org.apache.iceberg.Schema, org.apache.parquet.ParquetReadOptions, java.util.function.Function<org.apache.parquet.schema.MessageType, org.apache.iceberg.parquet.ParquetValueReader<?>>, org.apache.iceberg.mapping.NameMapping, org.apache.iceberg.expressions.Expression, boolean, boolean)" \
>           --new "method void org.apache.iceberg.parquet.ParquetReader<T>::<init>(org.apache.iceberg.io.InputFile, org.apache.iceberg.Schema, org.apache.parquet.ParquetReadOptions, java.util.function.Function<org.apache.parquet.schema.MessageType, org.apache.iceberg.parquet.ParquetValueReader<?>>, org.apache.iceberg.mapping.NameMapping, org.apache.iceberg.expressions.Expression, boolean, boolean, boolean)"
>     * All breaks in this project:

The CI failed because some changes in this PR breaks the API compatibility check.  
Although these APIs are public, they are only used in the iceberg-parquet project, 
not used in engine specific readers. Can I expect the same behavior from those  external 
connectors not in this repo? So I can make the revapi check accept these changes. Or should 
I make this change by deprecating  and adding a new method?